src/Core/Repositories/ExpandedPackageRepository.cs

```diff
@@ -117,9 +125,24 @@ private static string GetPackageRoot(string packageId, SemanticVersion version)
 
         private IPackage GetPackageInternal(string packageId, SemanticVersion version)
         {
-            var packagePath = GetPackagePath(packageId, version);
-            var manifestPath = Path.Combine(GetPackageRoot(packageId, version), packageId + Constants.ManifestExtension);
-            return new ZipPackage(() => _fileSystem.OpenFile(packagePath), () => _fileSystem.OpenFile(manifestPath));
+            var manifestPath = string.Empty;
+            try
+            {
+                var packagePath = GetPackagePath(packageId, version);
+                manifestPath = Path.Combine(GetPackageRoot(packageId, version), packageId + Constants.ManifestExtension);
```

Comment:
this needs a bug, because it is going to get confused if the nuspec name doesn't match the package id. We know we allow it on the server and in other scenarios.